tkg/client/upgrade_region.go

```diff
@@ -147,6 +147,11 @@ func (c *TkgClient) UpgradeManagementCluster(options *UpgradeClusterOptions) err
 	if err != nil {
 		return errors.Wrap(err, "error waiting for provider components to be up and running after upgrading them")
 	}
+
+	// Ensure Cluster API Provider AWS is running on the control plane before continuing with EC2 instance profile
+	if err := regionalClusterClient.PatchClusterAPIAWSControllersToUseEC2Credentials(); err != nil {
```

Comment:
I think this should execute before `c.WaitForProviders(regionalClusterClient, waitOptions)`

Reply:
Should I modify the mgmt cluster creation as well? Since on creation the patch is updated after provider is ready
https://github.com/vmware-tanzu/tanzu-framework/blob/fe8dd986a4313e34518dcea3d7025c3b8c2e84e0/tkg/client/init.go#L378-L384
https://github.com/vmware-tanzu/tanzu-framework/blob/fe8dd986a4313e34518dcea3d7025c3b8c2e84e0/tkg/client/init.go#L739

I added here because I suspect that upgrade goes into multi goroutine and cannot guarantee that the provider is ready  to be patched after `DoProvidersUpgrade`. Like the patched deployment got flushed out by the provider upgrade. If it can be ready to be patched, then it make no sense to have `WaitForProviders` here.

Reply:
We need to wait til deployment replica is greater than 1 when we patch CAPA deployment. @randomvariable What do you think?

Reply:
After sync with @DanielXiao , we found that there is no goroutine leak after `DoProvidersUpgrade` is finished. And `DoProvidersUpgrade` itself will make sure the provider is deployed successfully. So we decide to move `PatchClusterAPIAWSControllersToUseEC2Credentials` before `WaitForProviders` to reuse `WaitForProviders` for checking the patch result.